ui/src/pages/namespace/Permissions/components/PermisionsSelector/Row.tsx

```diff
@@ -0,0 +1,45 @@
+import { TableCell, TableRow } from "~/design/Table";
+
+import { Checkbox } from "~/design/Checkbox";
+import { joinPermissionString } from "./utils";
+
+type PermissionRowProps = {
+  resource: string;
+  scopes: string[];
+  availableScopes: string[];
+  selectedPermissions: string[];
+  onCheckedChange: (permissionValue: string, isChecked: boolean) => void;
+};
+
+export const PermissionRow = ({
+  resource,
+  scopes,
+  availableScopes,
+  selectedPermissions,
+  onCheckedChange,
+}: PermissionRowProps) => (
+  <TableRow>
+    <TableCell className="grow">{resource}</TableCell>
+    {availableScopes.map((availableScope) => {
+      const permissionString = joinPermissionString(availableScope, resource);
+      return (
+        <TableCell key={availableScope} className="px-2">
+          <div className="flex justify-center">
+            {scopes.includes(availableScope) ? (
+              <Checkbox
+                checked={selectedPermissions.includes(permissionString)}
+                onCheckedChange={(checked) => {
+                  if (checked !== "indeterminate") {
```

Comment:
You are referring to the `indeterminate` state, right? 

This condition is only for type safety. Since we are in control of this checkbox, we know that it can never be `indeterminate`.  However, Radix types the checked state as `boolean | 'indeterminate';` to [allow you to build a  checkbox component with three states](https://www.radix-ui.com/primitives/docs/components/checkbox#indeterminate)

Reply:
Ah, I see. I thought "indeterminate" was something from our backend. All good then.